Framework/Algorithms/test/PolarizationCorrectionWildesTest.h

```diff
@@ -1475,6 +1378,152 @@ class PolarizationCorrectionWildesTest : public CxxTest::TestSuite {
     }
     throw std::invalid_argument("Unknown polarization string: " + pol);
   }
+
+  std::pair<MatrixWorkspace_sptr, MatrixWorkspace_sptr> setupFailureWorkspaceAndEfficiencies() {
+    BinEdges edges{0.3, 0.6, 0.9, 1.2};
+    Counts counts{0., 0., 0.};
+
+    auto const ws00 = createWorkspace(1, counts);
+    const std::string wsName{"ws00"};
+    addWorkspaceToService(wsName, ws00);
+
+    auto effWS = idealEfficiencies(edges);
+
+    return {ws00, effWS};
+  }
+
+  void prepareWorkspacesAndRunCorrectionWithSampleState(const Counts &counts, const BinEdges &edges, size_t numClones,
+                                                        bool addSpinStateToLog,
+                                                        const std::vector<std::string> &expectedLogValues,
+                                                        const std::string &grouping = "") {
+    constexpr size_t nHist{2};
+    auto wsList = createWorkspaceList(counts, numClones);
+    auto wsNames = generateWorkspaceNames(numClones);
+    prepareAndRegisterWorkspaces(wsNames, wsList, nHist);
+
+    const auto effWS = efficiencies(edges);
+    auto alg = createWildesAlg(wsNames, effWS, grouping);
+    alg->setProperty("AddSpinStateToLog", addSpinStateToLog);
+
+    const auto outputWS = runAlg(std::move(alg));
+    validateNumberOfEntries(outputWS, numClones);
+
+    if (addSpinStateToLog) {
+      validateSpinStateLogs(outputWS, expectedLogValues);
+    } else {
+      validateNoSpinStateLogs(outputWS, numClones);
+    }
+  }
+
+  MatrixWorkspace_sptr createWorkspace(size_t nHist = 2, std::optional<BinEdges> edges = std::nullopt,
+                                       std::optional<Counts> counts = std::nullopt) {
+    if (!edges) {
+      edges = BinEdges{0.3, 0.6, 0.9, 1.2};
+    }
+
+    const auto yVal = 2.3;
+    if (!counts) {
+      counts = Counts{yVal, 4.2 * (yVal), yVal};
+    }
+
+    return create<Workspace2D>(nHist, Histogram(*edges, *counts));
+  }
+
+  MatrixWorkspace_sptr createWorkspace(size_t nHist, const Counts &counts) {
+    return createWorkspace(nHist, std::nullopt, counts);
+  }
+
+  MatrixWorkspace_sptr createWorkspace(const Counts &counts) { return createWorkspace(2, std::nullopt, counts); }
```

Comment:
```suggestion
  MatrixWorkspace_sptr createWorkspace(size_t nHist = 2, std::optional<Counts>& counts = std::nullopt,  std::optional<BinEdges>& edges = std::nullopt) {
    if (!edges) {
      edges = BinEdges{0.3, 0.6, 0.9, 1.2};
    }

    const auto yVal = 2.3;
    if (!counts) {
      counts = Counts{yVal, 4.2 * (yVal), yVal};
    }

    return create<Workspace2D>(nHist, Histogram(*edges, *counts));
  }

```
I would keep only one definition of this function and reverse the parameters count and edge in the function signature. I think you call it in a few places using only the count parameter, but you need to pass a number anyway to create a workspaces and it may not be always numHist =2 ,  so better pass it when you call the function.